base/libc.jl

```diff
@@ -77,6 +77,18 @@ flush_cstdio() = ccall(:jl_flush_cstdio, Void, ())
 @unix_only systemsleep(s::Real) = ccall(:usleep, Int32, (UInt32,), round(UInt32,s*1e6))
 @windows_only systemsleep(s::Real) = (ccall(:Sleep, stdcall, Void, (UInt32,), round(UInt32,s*1e3)); return Int32(0))
 
+type TimevalStruct
+   sec::Clong   # time_t
```

Comment:
I'm worried that we can't portably assume that `time_t` is a `Clong`.  Technically, it may not even be an integer.   http://stackoverflow.com/questions/471248/what-is-ultimately-a-time-t-typedef-to

Maybe something in the Makefile can print out `sizeof(time_t)` etc by compiling a little C program?